cider/exec/nextgen/jitlib/base/JITTuple.h

```diff
@@ -64,6 +65,44 @@ class JITTuple final : public JITBaseValue {
   boost::container::small_vector<JITBaseValue*, DefaultVectorSize> children;
 };
 
+class JITExprValue {
+ public:
+  JITExprValue() = default;
+  ~JITExprValue() = default;
+  JITExprValue(const JITExprValue&) = delete;
+  JITExprValue(JITExprValue&&) = delete;
+
+  template <typename T>
+  JITExprValue(T&& ptrs, bool is_variadic = false)
+      : ptrs_(std::forward<T>(ptrs)), is_variadic_(is_variadic) {}
+
+  // convert JITValuePointer to JITExprValue
+  JITExprValue(JITValuePointer&& val) {
+    is_variadic_ = false;
+    ptrs_.push_back(std::move(val));
+  }
+
+  cider::jitlib::JITValue& get_value() {
+    CHECK_GT(ptrs_.size(), 0);
+    return *ptrs_[0];
+  }
+  cider::jitlib::JITValue& get_null() {
+    CHECK_GT(ptrs_.size(), 1);
+    return *ptrs_[1];
+  }
+  cider::jitlib::JITValue& get_len() {
```

Comment:
Maybe the name style should be lower camel case.

Reply:
fixed